ocaml/typing/typecore.ml

```diff
@@ -5466,7 +5516,8 @@ and type_expect_
           simple_pat_mode mode, mode_default mode
         | Local_tuple arity ->
           let modes = List.init arity (fun _ -> Value.newvar ()) in
-          let mode = Value.regional_to_local (Value.join modes) in
+          let mode, _ = Value.newvar_above (Value.join (List.map Value.disallow_right modes)) in
```

Comment:
Why the change here?

Reply:
Because `join` returns a left-mode, but we need to construct an expected_mode, which is right-mode.